### PR TITLE
[fix](function) fix error when use  negative number in explode_numbers 

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

before :
negative number will core

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by Mryange** [comment]
run external

**Comment by Mryange** [comment]
run buildall

File: regression-test/suites/correctness/test_explode_numbers.groovy [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_explode_numbers") {
+    sql 'set enable_nereids_planner=true'
+    qt_select1 """
+        select e1 from (select 1 k1) as t lateral view explode_numbers(5) tmp1 as e1;
```
Comment (yiguolei) [inline regression-test/suites/correctness/test_explode_numbers.groovy:21]: ADD ORDER BY clause to  make sure the result is stable

**Comment by Mryange** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Review by Gabriel39 — DISMISSED** [review]

**Comment by Mryange** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**